### ISSUE TITLE
Pre-commit hook causes CMake error

### ISSUE THREAD

**Issue by neobrain** [citra-emu/citra#926]
The pre-commit hook introduced in  772ef097ead8d6be77cf00d753e8380fcac96710 is causing issues for me. Specifically, every time the CMakeLists.txt is processed with a pre-commit file already existing (e.g. because CMakeLists.txt had already been processed before, and is being re-read as part of calling `make`), I get the following error:

    CMake Error at CMakeLists.txt:9 (file):
      file COPY cannot set permissions on
      "~/citra/.git/hooks/pre-commit"

Further info: Upon trying to `rm` the file, a confirmation question is asked saying `rm: remove write-protected regular file ‘/home/tony/citra4/.git/hooks/pre-commit’?`. Upon answering `y`, removing the file works, of course.

**Comment by neobrain** [comment]
CC @Lectem

**Comment by yuriks** [comment]
What's your CMake version?

**Comment by neobrain** [comment]
2.8.11

**Comment by neobrain** [comment]
Apparently this is caused because improper permissions are being set on the target file. The file ends up having no permission flags set at all other than the "executable" flag.

**Comment by Lectem** [comment]
Windows or Linux ? It is a bit weird because it should be ignored if a file is already present : 

    if(NOT EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/.git/hooks/pre-commit)

**Comment by yuriks** [comment]
My guess is that CMake checks for existence by trying to read the file (which fails since it has no read permissions.)
```
[18:25:59] <yuriks> COPY preserves source permissions, so I think what should be done is add +x to the template script and simply leave out the permissions part of the command
```

**Comment by Lectem** [comment]
Could you try adding  WORLD_READ WORLD_WRITE  to FILE_PERMISSIONS on https://github.com/citra-emu/citra/blob/master/CMakeLists.txt#L11 ?

**Comment by yuriks** [comment]
@Lectem: This will still leave the file with wonky permissions. I think just letting COPY copy the perms from the original file is a better choice.

**Comment by Lectem** [comment]
@yuriks Thing is, git only sets the +x flag, or we could use SETUID.

**Comment by neobrain** [comment]
To be honest, I find it somewhat questionable to just override whatever pre-commit file the user might have installed for himself. Maybe this feature should just be removed, and if anything instead be provided as an option to the user (which effectively ruins all benefits, because most people won't bother). This is the route that CMake itself is going though as outlined on http://www.cmake.org/Wiki/Git/Hooks#Setup .

**Comment by Lectem** [comment]
Well, the thing is, it shouldn't override the pre-commit hook if you have one installed. Since there is a check.

**Comment by neobrain** [comment]
I'm still not too thrilled about the approach, since there is no simple way to update precommit hooks which we installed, either.

**Comment by Lectem** [comment]
Well we could always have the first line to store a commit number or a version and check against it.
I think fixing the bug and this question are two different things though.

**Comment by JSFernandes** [comment]
Just made a PR to handle this! https://github.com/citra-emu/citra/pull/928

**Comment by neobrain** [comment]
Fixed by #928.
